farmdata2_modules/fd2_tabs/resources/FarmOSAPI.js

```diff
@@ -95,7 +95,22 @@ function getAllPages(endpoint, arr=[]) {
  * // Note that the Map will not be available until the then() executes.
  */
 function getIDToUserMap(){
-    return getMap('/user', 'uid', 'name')
+    return new Promise((resolve, reject) => {
```

Comment:
Seems like we would still want to use getMap here and in the .then just remove the anonymous user?  That would allow us to not duplicate code from the getMap function.  Is there a reason for not doing it that way?

Reply:
We tried to do something along the lines of getMap(...).then((resp) => {resp.delete('Anonymous')}) but this kept causing weird errors like "(...).delete is not a function" or an empty user dropdown inside of the Seeding Report.

Reply:
I think you want to do is process the map in the function declared in the then and then resolve(...) the map.  So something like:
```
...
getMap(xxxx)
.then((map) => {
  // remove the thing from map
  resolve(map)
}
...
```

Reply:
This is what I wrote: 
```
function getUserToIDMap(){
    return getMap('/user', 'name', 'uid')
    .then((map) => {
        map.delete('Anonymous')
        resolve(map)
    })
}
```
But this produced an error: 
![image](https://user-images.githubusercontent.com/31524934/181117653-f9c47691-27e9-4e1e-a93e-1353fa5781cc.png)

Reply:
Give something like this a try:
```
function getIDToUserMap(){
    //return getMap('/user', 'uid', 'name')
    return new Promise((resolve, reject) => {
        getMap('/user', 'uid', 'name')
        .then((map) => {
            map.delete(null);
            resolve(map)
        },
        (error) => {
          reject(error);
        })
    })
}
```
It follows the same pattern you had but doesn't repeat the construction of the map, making use of the getMap function instead.

Reply:
It generates the same network banner error.

Reply:
I just tried that function and the corresponding one modified for getUserToIDMap and it seems to be working fine.  Do you get that banner when the page is loaded? Or when you do something else?

Reply:
On page load, let me try something to see if I can make it work.

Reply:
It's working for me now after resetting the Drupal cache. Is there any chance that loading that page once caches that data?

Reply:
The script could have been cached and not updated when you changed it.  Though, I didn't have to clear the cache for my edits to take effect.  

Anyway, as long as its working now that's good.